src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.EasyRequest.cs

```diff
@@ -311,8 +311,8 @@ private void SetMultithreading()
 
             private void SetTimeouts()
             {
-                // Set timeout limit on the connect phase.
-                SetCurlOption(CURLoption.CURLOPT_CONNECTTIMEOUT_MS, int.MaxValue);
+                // Set timeout limit on the connect phase. curl has bug on ARM so use max - 1s.
+                SetCurlOption(CURLoption.CURLOPT_CONNECTTIMEOUT_MS, int.MaxValue - 1000);
```

Comment:
It's probably worth opening a bug against curl, even if we have a work around. Their bug tracker is [here](https://github.com/curl/curl/issues).

Reply:
I was thinking about it. I would need to probably isolate C repo and show behavior with some public server.